packages/@aws-cdk/aws-redshift/lib/database-query.ts

```diff
@@ -0,0 +1,103 @@
+import * as path from 'path';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as cdk from '@aws-cdk/core';
+import * as customresources from '@aws-cdk/custom-resources';
+import { Construct } from 'constructs';
+import { Cluster } from './cluster';
+import { DatabaseProps } from './database-props';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+export interface DatabaseQueryProps extends DatabaseProps {
+  readonly handler: string;
+  readonly properties?: {[key: string]: any};
```

Comment:
I'd enjoy a bit more strong typing in TypeScript.

Since this class is not exported, you should be able to use type unions and generics to make it easier on yourself. Be sure to share the interface definitions between the runtime and construct code.

Reply:
I forgot that I had switched how I was passing the handle properties (used to be through the Lambda environment variables which need to be serialized)! Took a crack at it, let me know what you think